app/lib/analyzer/backend.dart

```diff
@@ -63,10 +63,7 @@ class AnalysisBackend {
   /// Stores the analysis, and either creates or updates its parent
   /// [PackageAnalysis] and [PackageVersionAnalysis] records.
   Future storeAnalysis(Analysis analysis) async {
-    assert(analysis.id == null);
-    await db.commit(inserts: [analysis]);
-    assert(analysis.id != null);
-
+    analysis.id ??= new DateTime.now().millisecondsSinceEpoch;
```

Comment:
Thanks for the explanation, that makes perfect sense.

It doesn't matter if you call it outside a transaction or not, once you get new id's back from the datastore, it is guaranteed they are not returned again. So in worst case there are just a few ids which got returned but are unused(e.g. when rolling back the inserting transaction).